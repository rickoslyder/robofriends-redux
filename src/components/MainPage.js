// import React from "react";
// import Header from "./Header";
// import SearchBox from "./SearchBox";
// import Scroll from "./Scroll";
// import CardList from "./CardList";

// const MainPage = () => {
//     return (
//         <>
//         <Header />
//         <SearchBox searchChange={onSearchChange}/>
//         <Scroll>
//           <CardList robots={filteredRobots} />
//         </Scroll>
//         </>
//     )
// }

import React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import { requestRobots, setSearchField } from '../actions';
import Header from '../components/Header';
import './MainPage.css';

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots()
  }

    filterRobots = robots => {
    return robots.filter(robot =>{
        return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        })
    }

  render() {
    const { searchField, onSearchChange, robots, isPending } = this.props;
    return isPending ?
      <h1>Loading</h1> :
      (
        <div className='tc'>
          <Header />
          <SearchBox searchChange={onSearchChange}/>
          <Scroll>
          <CardList robots={this.filterRobots(robots)} />
          </Scroll>
        </div>
      );
  }
}

export default MainPage