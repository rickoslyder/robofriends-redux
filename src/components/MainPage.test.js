import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import MainPage from './MainPage';

const mockStore = {
    robots: [],
    searchField: ''
}

const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false
}

const mockPropsPopulated = {
    onRequestRobots: jest.fn(),
    robots: [{
        id: 1,
        name: 'David Beckham',
        email: 'posh@spiceandbecks.com'
    },
    {
        id: 2,
        name: 'Leanne Graham',
        email: 'leanne@graham.com'
    },
    {
        id: 3,
        name: 'Richard the III',
        email: 'royalty@deadmonarch.com'
    },
    {
        id: 4,
        name: 'Leanne the III',
        email: 'fakeroyalty@deadmonarch.com'
    }
],
    searchField: 'Leanne',
    isPending: false
}

const mockMainPageComponent = <MainPage {...mockProps} />
const mockPopulatedMainPageComponent = <MainPage {...mockPropsPopulated} />

test('expect to render MainPage component', () => {
    render(mockMainPageComponent)
})

test('expect mock MainPage component JSX to match snapshot', () => {
    expect(mockMainPageComponent).toMatchSnapshot()
})

test('expect rendered mock MainPage component to match snapshot', () => {
    expect(render(mockMainPageComponent).baseElement).toMatchSnapshot()
})

test('filters robots correctly', () => {
    const user = userEvent.setup()
    render(mockPopulatedMainPageComponent)
    // look for all elements with 'Leanne ' - space included to ignore email
    const robots = screen.getAllByText('Leanne ', {exact: false})
    expect(robots).toBeDefined();
    // two Leannes - Leanne Graham and Leanne the III
    expect(robots.length).toEqual(2);
    expect(robots[0]).toHaveTextContent("Leanne Graham")
})

