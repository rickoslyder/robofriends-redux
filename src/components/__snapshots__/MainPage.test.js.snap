// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect mock MainPage component JSX to match snapshot 1`] = `
<MainPage
  isPending={false}
  onRequestRobots={[MockFunction]}
  robots={Array []}
  searchField=""
/>
`;

exports[`expect rendered mock MainPage component to match snapshot 1`] = `
<body>
  <div>
    <div
      class="tc"
    >
      <div>
        <button
          color="red"
        >
          Count: 
          0
        </button>
        <h1
          class="f1"
        >
          RoboFriends
        </h1>
      </div>
      <div
        class="pa2"
      >
        <input
          class="pa3 ba b--green bg-lightest-blue"
          placeholder="search robots"
          type="search"
        />
      </div>
      <div
        style="overflow: scroll; border: 5px solid black; height: 800px;"
      >
        <div />
      </div>
    </div>
  </div>
</body>
`;
